x-pack/plugins/monitoring/server/lib/alerts/fetch_missing_data.ts

```diff
@@ -0,0 +1,295 @@
+              },
+              aggs: {
+                es_uuids: {
+                  terms: {
+                    field: 'node_stats.node_id',
+                    size,
+                  },
+                  aggs: subAggs,
+                },
+                kibana_uuids: {
+                  terms: {
+                    field: 'kibana_stats.kibana.uuid',
+                    size,
+                  },
+                  aggs: subAggs,
+                },
+                beats: {
+                  filter: {
+                    bool: {
+                      must_not: {
+                        term: {
+                          'beats_stats.beat.type': 'apm-server',
+                        },
+                      },
+                    },
+                  },
+                  aggs: {
+                    beats_uuids: {
+                      terms: {
+                        field: 'beats_stats.beat.uuid',
+                        size,
+                      },
+                      aggs: subAggs,
+                    },
+                  },
+                },
+                apms: {
+                  filter: {
+                    bool: {
+                      must: {
+                        term: {
+                          'beats_stats.beat.type': 'apm-server',
+                        },
+                      },
+                    },
+                  },
+                  aggs: {
+                    apm_uuids: {
+                      terms: {
+                        field: 'beats_stats.beat.uuid',
+                        size,
+                      },
+                      aggs: subAggs,
+                    },
+                  },
+                },
+                logstash_uuids: {
+                  terms: {
+                    field: 'logstash_stats.logstash.uuid',
+                    size,
+                  },
+                  aggs: subAggs,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const response = await callCluster('search', params);
+  const indexBuckets = get(response, 'aggregations.index.buckets', []) as IndexBucketESResponse[];
+  const uniqueList: {
+    [id: string]: AlertMissingData;
+  } = {};
+  for (const indexBucket of indexBuckets) {
+    const clusterBuckets = indexBucket.clusters.buckets;
+    for (const clusterBucket of clusterBuckets) {
+      const clusterUuid = clusterBucket.key;
+      const uuidBuckets = findNonEmptyBucket(clusterBucket).buckets;
+      for (const uuidBucket of uuidBuckets) {
+        const stackProductUuid = uuidBucket.key;
+        const stackProduct = getStackProductFromIndex(indexBucket.key, clusterBucket);
+        const differenceInMs = +new Date() - uuidBucket.most_recent.value;
+        let stackProductName = stackProductUuid;
+        for (const nameField of nameFields) {
+          stackProductName = get(uuidBucket, `top.hits.hits[0]._source.${nameField}`);
```

Comment:
```suggestion
          stackProductName = get(uuidBucket, `document.hits.hits[0]._source.${nameField}`);
```
There was no `top.*` field name in my results, so I assume you wanted the above?

Reply:
Yes, thank you!